plugins/tray/trayplugin.cpp

```diff
@@ -343,40 +345,44 @@ void TrayPlugin::sniItemsChanged()
             m_passiveSNITrayMap.take(itemKey)->deleteLater();
         }
     }
-
-    m_registerSNIPID.clear();
-    QDBusConnection conn = QDBusConnection::connectToBus(QDBusConnection::SessionBus, "org.freedesktop.DBus");
     for (int i = 0; i < sinTrayKeyList.size(); ++i) {
         QString dbusName = SNITrayWidget::serviceAndPath(itemServicePaths.at(i)).first;
-        m_registerSNIPID.insert(conn.interface()->servicePid(dbusName));
-        traySNIAdded(sinTrayKeyList.at(i), itemServicePaths.at(i));
+        uint pid = conn.interface()->servicePid(dbusName);
```

Comment:
Like the XEmbedTrayWidget::getWindowPID, should get the pid of the dbus item at there by the SNITrayWidget::xxxx.